File: web/src/main/vite-project/src/kostratypes/uiDataVm.ts
import KostraFormTypeVm from "./kostraFormTypeVm"

interface UiDataVm {
    readonly releaseVersion: NonNullable<string>
    readonly years: NonNullable<number[]>
    readonly formTypes: NonNullable<KostraFormTypeVm[]>
}

export default UiDataVm

File: web/src/main/kotlin/no/ssb/kostra/web/viewmodel/UiDataVm.kt
package no.ssb.kostra.web.viewmodel

import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class UiDataVm(
    val releaseVersion: String,
    val years: Collection<Int>,
    val formTypes: Collection<KostraFormTypeVm>
)
